### PR TITLE
Features gui horizontal zooming

### PR DESCRIPTION
**Pull request by dvantetering**
This branch contains the "new" GUI. 
I seperated all GUI elements from the data set and restructured the Controllers, also the basic GUI elements have changed.

### CODE REVIEW

**Comment by arbreurkes** [comment]
This is basically the GUI 2.0 to continue building on.

File: src/main/java/application/controllers/MainController.java [hunk 1]
```diff
@@ -36,12 +40,61 @@ public MainController() {
...
+     */
+    public void createMenu() {
+        MenuFactory menuFactory = new MenuFactory(this);
+        menuBar = menuFactory.createMenu(menuBar);
+        this.getRoot().setTop(menuBar);
+    }
 
-        box.getChildren().addAll(screen, menuBar);
-        screen.getChildren().setAll();
+//    public void createSrollBar() {
```
Comment (ghost) [inline src/main/java/application/controllers/MainController.java:69]: Unused commented code. Candidate for removal?

File: src/main/java/application/controllers/MenuFactory.java [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+        Menu fileMenu = initFileMenu();
+        Menu viewMenu = initViewMenu();
+        Menu helpMenu = initHelpMenu();
+
+        bar.getMenus().addAll(fileMenu, viewMenu, helpMenu);
+        return bar;
+    }
+
+
+    private static Menu initHelpMenu() {
```
Comment (ghost) [inline src/main/java/application/controllers/MenuFactory.java:42]: No JDoc yet

File: src/main/java/application/controllers/ZoomController.java [hunk 3]
```diff
@@ -1,90 +1,58 @@
...
 
+        screenSize = Screen.getPrimary().getVisualBounds();
+
         zoomHandler = new ZoomHandler();
         keyHandler = new KeyHandler();
 
-        scaleTransform = new Scale(scaleValue, scaleValue, 0, 0);
-
-        this.setOnScroll(new ZoomHandler());
+//        this.setOnScroll(new ZoomHandler());
```
Comment (ghost) [inline src/main/java/application/controllers/ZoomController.java:35]: More commented code. Candidate for removal?

File: src/main/java/application/controllers/ZoomFactory.java [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+
+/**
+ * Created by Daphne van Tetering on 3-5-2016.
+ */
+public class ZoomFactory {
+
+    /**
+     * zoom factory oonstructor method.
+     */
+    public ZoomFactory() {
```
Comment (ghost) [inline src/main/java/application/controllers/ZoomFactory.java:14]: Constructor should be private, as ZoomFactory should probably be a static class.

File: src/main/java/application/controllers/ZoomFactory.java [hunk 5]
```diff
@@ -0,0 +1,30 @@
+package application.controllers;
+
+import javafx.scene.Group;
+import javafx.scene.control.ScrollPane;
+
+/**
+ * Created by Daphne van Tetering on 3-5-2016.
+ */
+public class ZoomFactory {
```
Comment (ghost) [inline src/main/java/application/controllers/ZoomFactory.java:9]: ZoomFactory can be made into a static class.

File: src/main/java/application/controllers/Zoomer.java [hunk 6]
```diff
@@ -0,0 +1,124 @@
...
+
+        zoomBox = new ZoomBox();
+        keyHandler = new KeyHandler();
+        borderPane = new BorderPane();
+
+        borderPane.setCenter(zoomPane);
+        borderPane.setBottom(zoomBox.getZoomBox());
+        borderPane.setOnKeyPressed(keyHandler);
+
+//        zoomPane.widthProperty().addListener(event -> bounds());
```
Comment (ghost) [inline src/main/java/application/controllers/Zoomer.java:51]: More commented code, candidate for removal.

File: src/main/java/application/fxobjects/graph/MouseHandling.java [hunk 7]
```diff
@@ -12,57 +13,34 @@
     final DragContext dragContext = new DragContext();
 
     Graph graph;
-    EventHandler<MouseEvent> onMousePressedEventHandler = new EventHandler<MouseEvent>() {
```
Comment (ghost) [inline src/main/java/application/fxobjects/graph/MouseHandling.java:15]: No JDoc

File: src/main/java/application/fxobjects/graph/cell/BaseLayout.java [hunk 8]
```diff
@@ -34,31 +41,65 @@ public BaseLayout(Graph graph, int offset) {
     public void execute() {
         List<Cell> cells = graph.getModel().getAllCells();
 
+//        for (Cell cell : cells) {
```
Comment (ghost) [inline src/main/java/application/fxobjects/graph/cell/BaseLayout.java:44]: More commented code, candidate for removal.

File: src/main/java/core/graph/Graph.java [hunk 9]
```diff
@@ -0,0 +1,153 @@
...
+    /**
+     * Get the model of the Graph.
+     * @return The model of the graph.
+     */
+    public Model getModel() {
+        return model;
+    }
+
+
+    public void addGraphComponents() {
```
Comment (ghost) [inline src/main/java/core/graph/Graph.java:45]: No JDoc

**Comment by ric-jon** [comment]
This is a large pull request! Quite a lot of javadoc is missing.

**Pull request closed without merging**